libraries/BufferedBleSpp/nanoFramework/BufferedBleSpp.cs

```diff
@@ -0,0 +1,275 @@
+            {
+                UserDescription = "TX Amount Characteristic",
+                CharacteristicProperties =  GattCharacteristicProperties.Read
+            };
+
+            GattLocalCharacteristicResult txAmountCharRes = serviceProvider.Service.CreateCharacteristic(TxAmountCharacteristicUUID, txAmountParam);
+            if (txAmountCharRes.Error != nanoFramework.Device.Bluetooth.BluetoothError.Success)
+            {
+                throw new ArgumentException("Unable to create TX Amount Characteristic");
+            }
+
+            GattLocalCharacteristicParameters txBufferParam = new GattLocalCharacteristicParameters()
+            {
+                UserDescription = "TX Characteristic",
+                CharacteristicProperties = GattCharacteristicProperties.Read
+            };
+
+            GattLocalCharacteristicResult txBufferCharRes = serviceProvider.Service.CreateCharacteristic(TxBufferCharacteristicUUID, txBufferParam);
+            if (txBufferCharRes.Error != nanoFramework.Device.Bluetooth.BluetoothError.Success)
+            {
+                throw new ArgumentException("Unable to create TX Amount Characteristic");
+            }
+
+            txAmountCharacteristic = txAmountCharRes.Characteristic;
+            txAmountCharacteristic.ReadRequested += _txCharacteristic_AmountReadRequested;
+
+            txBufferCharacteristic = txBufferCharRes.Characteristic;
+            txBufferCharacteristic.ReadRequested += _txCharacteristic_BufferReadRequested;
+
+        }
+
+
+        /// <summary>
+        /// Start device advertising
+        /// </summary>
+        /// <param name="deviceName">Device name for Advertising</param>
+        /// <returns></returns>
+        public bool Start()
+        {
+            GattServiceProviderAdvertisingParameters advParameters = new GattServiceProviderAdvertisingParameters
+            {
+                DeviceName = deviceName,
+                IsDiscoverable = true,
+                IsConnectable = true
+            };
+
+            serviceProvider.StartAdvertising(advParameters);
+
+            return true;
+        }
+
+        /// <summary>
+        /// Stop advertising.
+        /// </summary>
+        public void Stop()
+        {
+            serviceProvider?.StopAdvertising();
+        }
+
+        /// <summary>
+        /// Event handler for Received data
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="WriteRequestEventArgs"></param>
+        private void RxAmountCharacteristic_WriteRequested(GattLocalCharacteristic sender, GattWriteRequestedEventArgs WriteRequestEventArgs)
+        {
+            GattWriteRequest request = WriteRequestEventArgs.GetRequest();
+
+            byte[] data = new byte[request.Value.Length];
+
+            DataReader rdr = DataReader.FromBuffer(request.Value);
+            rdr.ReadBytes(data);
+
+            #region *// Init the process

```

Comment:
one more time on regions, the comment is largely enough, no real need for a region